contrib/cloud.google.com/go/pubsub.v1/pubsub.go

```diff
@@ -0,0 +1,70 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+// Package pubsub provides functions to trace the cloud.google.com/pubsub/go package.
+package pubsub
+
+import (
+	"context"
+
+	"github.com/apex/log"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"cloud.google.com/go/pubsub"
+)
+
+// Publish publishes a message on the specified topic, waits for publishing to complete, and returns the result.
+// It is functionally equivalent to (*pubsub.Topic).Publish(ctx, msg).Get(ctx), but it also ensures that the datadog
+// tracing metadata is propagated as attributes attached to the published message.
+func Publish(ctx context.Context, t *pubsub.Topic, msg *pubsub.Message) (serverID string, err error) {
+	span, ctx := tracer.StartSpanFromContext(
+		ctx,
+		"pubsub.publish",
+		tracer.ResourceName(t.String()),
+		tracer.SpanType(ext.SpanTypeMessageProducer),
+		tracer.Tag("message_size", len(msg.Data)),
+		tracer.Tag("ordering_key", msg.OrderingKey),
+	)
+	defer func() {
+		span.Finish(tracer.WithError(err))
+	}()
+	if msg.Attributes == nil {
+		msg.Attributes = make(map[string]string)
+	}
+	ierr := tracer.Inject(span.Context(), tracer.TextMapCarrier(msg.Attributes))
+	if ierr != nil {
```

Comment:
```suggestion
	if err := tracer.Inject(span.Context(), tracer.TextMapCarrier(msg.Attributes)); err != nil {
```
No need for the odd variable name :)